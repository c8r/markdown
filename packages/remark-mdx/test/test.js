const unified = require('unified')
const remarkParse = require('remark-parse')
const remarkMDX = require('..')
const mdxAstToMdxHast = require('../../mdx/mdx-ast-to-mdx-hast')
const mdxHastToJsx = require('../../mdx/mdx-hast-to-jsx')

const FIXTURE = `
import Foo from './bar'
export { Baz } from './foo'
export default Foo

# Hello, world! <Foo bar={{ baz: 'qux' }} />

<Baz>
  Hi!
</Baz>
`

// Manually apply all mdx transformations for now
// so we don't introduce a circular dependency.
//
// When remark-mdx is used by core we can remove
// these integration tests
const transpile = mdx => {
  const result = unified()
    .use(remarkParse)
    .use(remarkMDX)
    .use(mdxAstToMdxHast)
    .use(mdxHastToJsx)
    .processSync(mdx)

  return result.contents
}

it('correctly transpiles', () => {
  const result = transpile(FIXTURE)

  expect(result).toMatchSnapshot()
})
