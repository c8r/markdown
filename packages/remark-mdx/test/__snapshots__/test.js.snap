// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly transpiles 1`] = `
"import Foo from './bar'

export { Baz } from './foo'

const layoutProps = {
  
};
export default class MDXContent extends React.Component {
  constructor(props) {
    super(props)
    this.layout = Foo
  }
  render() {
    const { components, ...props } = this.props

    return <MDXTag
             name=\\"wrapper\\"
             Layout={this.layout} layoutProps={Object.assign({}, layoutProps, props)}
             components={components}>


<MDXTag name=\\"h1\\" components={components}>{\`Hello, world! \`}<Foo bar={{ baz: 'qux' }} /></MDXTag>
<Baz>
  Hi!
</Baz>
           </MDXTag>
  }
}"
`;
